encoding/form/proto_decode.go

```diff
@@ -80,11 +80,18 @@ func populateFieldValues(v protoreflect.Message, fieldPath []string, values []st
 func getFieldDescriptor(v protoreflect.Message, fieldName string) protoreflect.FieldDescriptor {
 	fields := v.Descriptor().Fields()
 	var fd protoreflect.FieldDescriptor
-	if fd = getDescriptorByFieldAndName(fields, fieldName); fd == nil {
-		if v.Descriptor().FullName() == structMessageFullname {
+	fd = getDescriptorByFieldAndName(fields, fieldName)
+	if fd == nil {
+		reg := regexp.MustCompile(`^.*?\[`)
+		switch {
+		case v.Descriptor().FullName() == structMessageFullname:
 			fd = fields.ByNumber(structFieldsFieldNumber)
-		} else if len(fieldName) > 2 && strings.HasSuffix(fieldName, "[]") {
+		case len(fieldName) > 2 && strings.HasSuffix(fieldName, "[]"):
```

Comment:
The key is 'map[kratos]', instead of map(this field is Map), this field cannot be obtained